crates/iceberg/src/spec/manifest.rs

```diff
@@ -94,6 +94,12 @@ impl Manifest {
         &self.entries
     }
 
+    /// Consume this Manifest, returning its constituent parts
+    pub fn consume(self) -> (Vec<ManifestEntryRef>, ManifestMetadata) {
```

Comment:
We often use `into_parts` for API like this, do you like it?

Reply:
This change was no longer present in this PR after it got rebased, but I've renamed it anyway for consistency with the rest of the codebase. Thanks!